Improving guidelines on changelog (#3435)
Azure/communication-ui-library: docs/references/changelog-guidelines.md

# Changelog Guidelines

To ease our release process, we need to follow some guidelines when adding a new changelog entry.

> `rush changelog` will give developers the ooportunity to create a change file to describe your changes. 

Reviewer: 'oportunity' typo

The `comment` field is used to generate the changelog entry, so it's important to follow the guidelines below.

## Type
Select whether your change is a Major, Minor, or Patch (for more information on which to choose see [Semantic Versioning](https://semver.org/)).

When to use each type:

> 1. Prerelease: Changes that are not yet released to the public.

Reviewer 2: These types don't match the types in the paragraph above. Which is correct?

Author: Ah good catch the ones here are correct.

1. Patch: Bug fixes; no API changes.
1. Minor: Small feature; backwards compatible API changes.
1. None: This change does not affect the published package in any way.

## Area

The `area` field is used to group the changelog entries under specific categories.
Categories and their definitions: 
1. Bug Fix: Regression in functionality, or defect in existing functionality.
1. Feature: Net new functionality, or significant enhancement to existing functionality.
1. Improvement: Additional work on existing functionality outside of its initial dev cycle.

## Describe changes
Text field entry to describe the changes. The changelog entry to be in present tense, capitalized, and to NOT end with periods.

This is what will be included in the changelog when a new version is released, so keep it concise and meaningful. See [Tips for writing meaningful changelog entries](../reviewer-notes/tips-for-writing-changelog-entries.md).

### Examples
```md

> - Adds functionality to open image gallery when user taps on an inline image

Reviewer: ```suggestion
- Add functionality to open image gallery when user taps on an inline image
```

> - Starting new `workstream` or `feature` 

Reviewer: Should we have more strict rules about Ving, V and V in third person singular form?

Author: We could, would you like to suggest / update the examples on the verbiage to follow

> - Starting new `workstream` or `feature` 

Reviewer: ```suggestion
- Start new `workstream` or `feature` 
```

> - Adding to `workstream` or `feature` 

Reviewer: ```suggestion
- Add `functionality` to `workstream` or `feature` 
```

> - Fixes app crash when user taps on call button

Reviewer: ```suggestion
- Fix for app crash when user taps on call button
```

> - Improves performance of the app by reducing the number of API calls

Reviewer: ```suggestion
- Improve performance of the app by reducing the number of API calls
```
